### PR TITLE
Completed configuration for spring mvc in project

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class CinemaHallDaoImpl implements CinemaHallDao {
-    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
+    private static final Logger logger = Logger.getLogger(OrderDaoImpl.class);
+    private SessionFactory sessionFactory;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java]: Should be private final

File: src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+
+    @Autowired
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java]: you don't need this @Autowired with constructor

File: src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 import com.dev.cinema.exceptions.AuthenticationException;
-import com.dev.cinema.lib.Inject;
-import com.dev.cinema.lib.Service;
 import com.dev.cinema.model.User;
 import com.dev.cinema.util.HashUtil;
 import java.util.Optional;
+import org.springframework.stereotype.Component;
 
-@Service
+@Component
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java]: Why not - Service?

File: src/main/java/com/dev/cinema/service/CinemaHallServiceImpl.java [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
-import com.dev.cinema.lib.Inject;
-import com.dev.cinema.lib.Service;
 import com.dev.cinema.model.CinemaHall;
 import java.util.List;
+import org.springframework.stereotype.Service;
 
 @Service
 public class CinemaHallServiceImpl implements CinemaHallService {
-    @Inject
     private CinemaHallDao cinemaHallDao;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/CinemaHallServiceImpl.java]: private final

File: src/main/java/com/dev/cinema/service/MovieServiceImpl.java [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
-import com.dev.cinema.lib.Inject;
-import com.dev.cinema.lib.Service;
 import com.dev.cinema.model.Movie;
 import java.util.List;
+import org.springframework.stereotype.Service;
 
 @Service
 public class MovieServiceImpl implements MovieService {
-    @Inject
     private MovieDao movieDao;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/MovieServiceImpl.java]: final...

File: src/main/java/com/dev/cinema/service/MovieSessionServiceImpl.java [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
-import com.dev.cinema.lib.Service;
 import com.dev.cinema.model.MovieSession;
 import java.time.LocalDate;
 import java.util.List;
+import org.springframework.stereotype.Service;
 
 @Service
 public class MovieSessionServiceImpl implements MovieSessionService {
-    @Inject
     private MovieSessionDao movieSessionDao;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/MovieSessionServiceImpl.java]: final?

File: src/main/java/com/dev/cinema/service/OrderServiceImpl.java [hunk 7]
```diff
@@ -1,22 +1,25 @@
...
 import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
+import org.springframework.stereotype.Service;
 
 @Service
 public class OrderServiceImpl implements OrderService {
-    @Inject
+
     private OrderDao orderDao;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/OrderServiceImpl.java]: final?

File: src/main/java/com/dev/cinema/service/OrderServiceImpl.java [hunk 8]
```diff
@@ -1,22 +1,25 @@
...
-    @Inject
     private ShoppingCartService shoppingCartService;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/OrderServiceImpl.java]: final?

File: src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java [hunk 9]
```diff
@@ -2,20 +2,23 @@
...
 import com.dev.cinema.model.User;
+import org.springframework.stereotype.Service;
 
 @Service
 public class ShoppingCartServiceImpl implements ShoppingCartService {
-    @Inject
+
     private TicketDao ticketDao;
-    @Inject
     private ShoppingCartDao shoppingCartDao;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java]: final = both?

File: src/main/java/com/dev/cinema/service/UserServiceImpl.java [hunk 10]
```diff
@@ -1,17 +1,20 @@
...
 import com.dev.cinema.model.User;
 import com.dev.cinema.util.HashUtil;
 import java.util.Optional;
+import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
-    @Inject
+
     private UserDao userDao;
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/service/UserServiceImpl.java]: final

**Review by fyarkiy — APPROVED** [review]
Good job!

File: src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java (hunk 1 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java (hunk 2 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java (hunk 3 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/CinemaHallServiceImpl.java (hunk 4 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/CinemaHallServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/MovieServiceImpl.java (hunk 5 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/MovieServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/MovieSessionServiceImpl.java (hunk 6 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/MovieSessionServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/OrderServiceImpl.java (hunk 7 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/OrderServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/OrderServiceImpl.java (hunk 8 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/OrderServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java (hunk 9 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/UserServiceImpl.java (hunk 10 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/UserServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java [hunk 11]
```diff
@@ -1,24 +1,29 @@
...
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class CinemaHallDaoImpl implements CinemaHallDao {
-    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
+    private static final Logger logger = Logger.getLogger(OrderDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
```

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 12]
```diff
@@ -1,24 +1,29 @@
...
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class MovieDaoImpl implements MovieDao {
-    private static final Logger logger = Logger.getLogger(MovieDaoImpl.class);
+    private static final Logger logger = Logger.getLogger(OrderDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(MovieDaoImpl.class);
```

File: src/main/java/com/dev/cinema/dao/MovieSessionDaoImpl.java [hunk 13]
```diff
@@ -1,24 +1,29 @@
...
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class MovieSessionDaoImpl implements MovieSessionDao {
-    private static final Logger logger = Logger.getLogger(MovieSessionDaoImpl.class);
+    private static final Logger logger = Logger.getLogger(OrderDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/MovieSessionDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(MovieSessionDaoImpl.class);
```

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java [hunk 14]
```diff
@@ -1,24 +1,29 @@
...
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class ShoppingCartDaoImpl implements ShoppingCartDao {
-    private static final Logger logger = Logger.getLogger(ShoppingCartDaoImpl.class);
+    private static final Logger logger = Logger.getLogger(UserDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(ShoppingCartDaoImpl.class);
```

File: src/main/java/com/dev/cinema/dao/TicketDaoImpl.java [hunk 15]
```diff
@@ -1,23 +1,28 @@
...
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class TicketDaoImpl implements TicketDao {
-    private static final Logger logger = Logger.getLogger(TicketDaoImpl.class);
+    private static final Logger logger = Logger.getLogger(UserDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/TicketDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(TicketDaoImpl.class);
```

File: src/main/java/com/dev/cinema/service/OrderServiceImpl.java [hunk 16]
```diff
@@ -1,21 +1,24 @@
...
 import java.util.List;
+import org.springframework.stereotype.Service;
 
 @Service
 public class OrderServiceImpl implements OrderService {
-    @Inject
-    private OrderDao orderDao;
-    @Inject
-    private ShoppingCartService shoppingCartService;
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/service/OrderServiceImpl.java]: let's remove empty line

File: src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java [hunk 17]
```diff
@@ -2,19 +2,22 @@
...
 import com.dev.cinema.model.User;
+import org.springframework.stereotype.Service;
 
 @Service
 public class ShoppingCartServiceImpl implements ShoppingCartService {
-    @Inject
-    private TicketDao ticketDao;
-    @Inject
-    private ShoppingCartDao shoppingCartDao;
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java]: same here

File: src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java (hunk 11 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/CinemaHallDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java (hunk 12 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java (hunk 14 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/TicketDaoImpl.java (hunk 15 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/TicketDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/service/OrderServiceImpl.java (hunk 16 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/service/OrderServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/MovieSessionDaoImpl.java (hunk 13 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/MovieSessionDaoImpl.java]: fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**